pkg/api/latest/latest.go

```diff
@@ -99,8 +100,24 @@ func InterfacesFor(version string) (*meta.VersionInterfaces, error) {
 }
 
 func init() {
+	// Env var KUBE_API_VERSIONS is a comma separated list of supported API versions.
+	// The versions should be in the order of most preferred to the least.
+	supportedVersions := os.Getenv("KUBE_API_VERSIONS")
+	if supportedVersions == "" {
+		supportedVersions = "v1beta3,v1beta1,v1beta2,v1"
+	}
+	versions := strings.Split(supportedVersions, ",")
+	// The first version in the list is the latest version.
+	Version = versions[0]
+	Codec = runtime.CodecFor(api.Scheme, Version)
+	// Put the versions in Versions in reverse order.
+	Versions = []string{}
+	for i := len(versions) - 1; i >= 0; i-- {
```

Comment:
It's probably worth checking and panicing (w/ helpful error message) if any of these are empty-- that would cause us to serialize the internal version.